ui/app/ducks/swaps/swaps.js

```diff
@@ -530,3 +613,47 @@ export const signAndSendTransactions = (history, metaMetricsEvent) => {
     await forceUpdateMetamaskState(dispatch)
   }
 }
+
+export function fetchMetaSwapsGasPriceEstimates () {
+  return async (dispatch, getState) => {
+    const state = getState()
+    const priceEstimatesLastRetrieved = getSwapsPriceEstimatesLastRetrieved(state)
+    const timeLastRetrieved = priceEstimatesLastRetrieved || loadLocalStorageData('METASWAP_GAS_PRICE_ESTIMATES_LAST_RETRIEVED') || 0
+
+    dispatch(swapGasPriceEstimatesFetchStarted())
+
+    let priceEstimates
+    try {
+      if (Date.now() - timeLastRetrieved > 30000) {
+        priceEstimates = await fetchSwapsGasPrices()
+      } else {
+        const cachedPriceEstimates = loadLocalStorageData('METASWAP_GAS_PRICE_ESTIMATES')
+        priceEstimates = cachedPriceEstimates || await fetchSwapsGasPrices()
+      }
+    } catch (e) {
```

Comment:
Ignoring both of these errors is a bit concerning. Printing it to the console would be a little informative at least.

We might also want to re-throw if it's something unexpected. I think `NetworkError` and `Fetch failed with status: *` are the only two failures we should expect here (for network errors and failed HTTP responses respectively). Anything else is unexpected.

Reply:
Additionally, a `log.warn` might be helpful here? So that if this is encountered, the console will inform that this rare-but-not-unexpected branch was reached.

Reply:
Addressed in 8209364c4